### ISSUE TITLE
Problems generating help in system with multiple Python installations

### ISSUE THREAD

**Issue by heplesser** [nest/nest-simulator#678]
André Sevenius reported the following problem on NEST User 14 March 2017:

I get the following error:

```
Extracting help information; this may take a litte while.
Traceback (most recent call last):
  File "/opt/rh/anaconda/lib/python3.6/site-packages/site.py", line 73, in <module>
    __boot()
  File "/opt/rh/anaconda/lib/python3.6/site-packages/site.py", line 3, in __boot
    import os
  File "/opt/rh/anaconda/lib/python3.6/os.py", line 409
    yield from walk(new_path, topdown, onerror, followlinks)
             ^
SyntaxError: invalid syntax
make[2]: *** [generate_help] Error 1
make[1]: *** [doc/CMakeFiles/generate_help.dir/all] Error 2
make: *** [all] Error 2
```

```
NEST Configuration Summary
--------------------------------------------------------------------------------
Build type          : 
Target System       : Linux
Cross Compiling     : FALSE
C compiler          : /bin/cc
C compiler flags    :  -O2 -Wall -fopenmp 
C++ compiler        : /bin/c++
C++ compiler flags  :  -O2 -Wall -fopenmp 
Build dynamic       : ON

Built-in modules    : models;precise;topology
User modules        : None
Python bindings     : Yes (Python 3.6: /opt/rh/anaconda/bin/python3.6m)
       Includes     : /opt/rh/anaconda/include/python3.6m
       Libraries    : /opt/rh/anaconda/lib/libpython3.6m.so

Cython bindings     : Yes (Cython 0.25.2: /opt/rh/anaconda/bin/cython)
Use threading       : Yes (OpenMP: -fopenmp)
Use GSL             : Yes (GSL 1.15)
    Includes        : /usr/include
    Libraries       : /usr/lib64/libgsl.so;/usr/lib64/libgslcblas.so

Use Readline        : Yes (GNU Readline 6.2)
    Includes        : /usr/include
    Libraries       : /lib64/libreadline.so;/lib64/libncurses.so

Use libltdl         : Yes (LTDL 2.4.2)
    Includes        : /usr/include
    Libraries       : /lib64/libltdl.so

Use doxygen         : Yes (/bin/doxygen)
                    : target `doc` available
Use MPI             : No
Use MUSIC           : No
Use libneurosim     : No
```

So as you can see, python should be setup correctly and it finds the right inerpreter. But this is a classical error where python3.x is interpreted by 2.7, but I've checked paths and everything, even changing aliases to make sure no reference to python2 exist.

**Comment by heplesser** [comment]
@jougs Assigning to you as this is a follow-up to #647.

Suggested work-around (not tested): In file `doc/CMakeLists.txt` delete lines 20-44
```cmake
if ( NOT CMAKE_CROSSCOMPILING ) 
  ... 
endif()
```
Online help will not be available in this case.

I suspect this problem is due to the fact that we have hardcoded python on lines 26-27

```cmake
  COMMAND python -B generate_help.py "${PROJECT_SOURCE_DIR}" "${PROJECT_BINARY_DIR}"
  COMMAND python -B generate_helpindex.py "${PROJECT_BINARY_DIR}/doc"
```

and line 40

```cmake
  COMMAND python -B generate_helpindex.py \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DOCDIR}\"
```

Should we get python from the CMake-Variable containing the Python excutable CMake has discovered?

**Comment by heplesser** [comment]
See also #712 on what happens if no Python is available at all.

**Comment by heplesser** [comment]
@steffengraber Together with #712, I think the situation is as follows:
- If no Python is available at all, or if Python is older than 2.7.8 (`re`-bug), we skip help generation completely.
- Otherwise, we need to use the Python that CMake has discovered instead of hard-coding `python` as is the case at present.

**Comment by steffengraber** [comment]
@heplesser Can we close this issue?

**Comment by heplesser** [comment]
Fixed by #715.
